### PR TITLE
Dependencies and format

### LINKED ISSUE #63: npm audit failes with 5 vulnerabilities (4 low, 1 moderate)
This plugin references outdated libraries with vulnerabilities. They can be listeted with

`npm audit cordova-plugin-localization-strings`

which gives the audit report below. I'll have a look at it and try to update the references.

```
# npm audit report

xmldom  *
Severity: moderate
Misinterpretation of malicious XML input - https://github.com/advisories/GHSA-h6q6-9hqw-rwfv
Misinterpretation of malicious XML input - https://github.com/advisories/GHSA-5fg8-2547-mr8q
fix available via `npm audit fix --force`
Will install cordova-plugin-localization-strings@2.0.0, which is a breaking change
node_modules/xmldom
  cordova-plugin-localization-strings  *
  Depends on vulnerable versions of xcode
  Depends on vulnerable versions of xmldom
  node_modules/cordova-plugin-localization-strings
  plist  0.3.2 - 3.0.1
  Depends on vulnerable versions of xmldom
  node_modules/simple-plist/node_modules/plist
    simple-plist  <=0.3.0
    Depends on vulnerable versions of plist
    node_modules/simple-plist
      xcode  0.8.3 - 1.1.0
      Depends on vulnerable versions of simple-plist
      node_modules/xcode

5 vulnerabilities (4 low, 1 moderate)
```

### PR DESCRIPTION
**Pull request by ath0mas**
More dependencies update (fix #63): correct version in package-lock, [xcode](https://www.npmjs.com/package/xcode) v3, [@xmldom/xmldom](https://www.npmjs.com/package/@xmldom/xmldom)
\+ Some formatting mostly to cleanup js whitespaces and indents

Used and everything is fine for both android and ios in my case 👍 

(you should toggle "Hide whitespace" or look at each commit separately for easier review)

---

Dependencies:
- `fs-extra`: replaced by node `fs` + smaller single-responsibility `mkdirp`
- `iconv-lite`: removed, node 'utf8' default encoding to write files is fine
- `lodash`: replaced by `underscore`, same api, smaller bundle
- `xcode`: updated from v0.9 to v3
- `xml-writer`removed, unused
- `xml2js`: ok
- `xmldom`: removed, unique use replaced by regex matching
- `path`: removed, node `path` is fine
- `glob`: ok

Still OK for both android and ios in my test project.

Maybe add a line inside Readme about "new" node version "compatibility" in 4.0.0? reviewing package-lock.json `"engines"` may help.
=> `"node": ">=10"` for `mkdirp` and `xcode` v3

### CODE REVIEW

**Comment by rodrigograca31** [comment]
I will merge and publish as 4.0.0 as soon as I solve an issue with my npm account which isnt letting me publish... :sob:

**Comment by ath0mas** [comment]
I will have some more cleanup and minor rewrites to suggest later today ; in this PR or a new one after this one merged?

**Comment by rodrigograca31** [comment]
I think its fine if you keep it in this one. like I said I will review and publish everything in the end as 4.0.0

**Comment by ath0mas** [comment]
Done with my changes - ready for review

**Comment by ath0mas** [comment]
Done with my changes - ready for review

**Comment by ath0mas** [comment]
Done for me - ready for review

File: scripts/create_android_strings.js [hunk 1]
```diff
@@ -196,7 +196,7 @@ function processResult(context, lang, langJson, stringXmlJson) {
     _.forEach(langJsonToProcess, function (val, key) {
         // positional string format is in Mac OS X format. change to android format
         val = val.replace(/\$@/gi, '$s');
-        val = val.replace(/\'/gi, "\\'");
+        val = val.replace(/'/gi, "\\'");
```
Comment (rodrigograca31) [inline scripts/create_android_strings.js:166]: I hope this doesnt break stuff. probably wont but I dont know regex well enought. 😂

File: scripts/create_strings.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-    if (platforms.indexOf('ios') >= 0) {
-        promises.push(ios_script(context));
-    }
-
     if (platforms.indexOf('android') >= 0) {
         promises.push(android_script(context));
     }
 
+    if (platforms.indexOf('ios') >= 0) {
```
Comment (rodrigograca31) [inline scripts/create_strings.js:13]: not sure why this re-order was necessary ... thinking

File: scripts/create_ios_strings.js [hunk 3]
```diff
@@ -43,12 +42,7 @@ function writeStringFile(plistStringJsonObj, lang, fileName) {
     fs.ensureDir(lProjPath, function (err) {
         if (!err) {
             var stringToWrite = jsonToDotStrings(plistStringJsonObj);
-            var buffer = iconv.encode(stringToWrite, 'utf8');
-
-            fs.open(lProjPath + '/' + fileName, 'w', function (err, fd) {
-                if (err) throw err;
-                fs.writeFileSync(fd, buffer);
-            });
+            fs.writeFileSync(path.join(lProjPath, fileName), stringToWrite);
```
Comment (rodrigograca31) [inline scripts/create_ios_strings.js:45]: I was wondering exactly this! I didnt understand why there was encoding going on.... thanks!

**Comment by rodrigograca31** [comment]
Let me ask you for just one last "fix"
Can you take a look at the use of `fs-extra` ? 

seems its used to use `fs.ensureDir` which as far as I know the other functions (`fs.readFile, fs.writeFile, etc`) are in "normal" `fs` (from node)... can you replace it and remove `fs-extra` ?

You did a great job so far! The community really appreciates it!

File: scripts/create_android_strings.js (hunk 1 above)
Comment (ath0mas) [reply scripts/create_android_strings.js:167]: My IDE detected it as a "Redundant character escape in RegExp" and indeed quotes are not reserved characters.

File: scripts/create_strings.js (hunk 2 above)
Comment (ath0mas) [reply scripts/create_strings.js:13]: Not necessary but alpha order as much as common order in Cordova plugins to deal with Android before iOS.

File: package.json [hunk 4]
```diff
@@ -29,7 +29,6 @@
         "glob": "^7.2.0",
         "iconv-lite": "^0.6.3",
         "lodash": "^4.17.21",
-        "path": "^0.12.7",
```
[the remark this answers is not in the document]
Comment (ath0mas) [reply package.json]: Yes! :) and more after all ; I am going to edit PR description to give detail about what changed here with all those newer commits added.

File: scripts/create_ios_strings.js (hunk 3 above)
Comment (ath0mas) [reply scripts/create_ios_strings.js]: It seems that, before #35, the encoding used to write file was 'utf16' (requiring iconv-lite and buffer).
Back to using 'utf8' means we can simply use `writeFile` 👍 .

File: package.json [hunk 5]
```diff
@@ -29,7 +29,6 @@
         "glob": "^7.2.0",
         "lodash": "^4.17.21",
         "xcode": "^3.0.1",
-        "xml-writer": "^1.7.0",
```
[the remark this answers is not in the document]
Comment (ath0mas) [reply package.json]: this one was never required nor used yes

**Comment by ath0mas** [comment]
> Let me ask you for just one last "fix" Can you take a look at the use of `fs-extra` ?
> 
> seems its used to use `fs.ensureDir` because as far as I know the other functions (`fs.readFile, fs.writeFile, etc`) are in "normal" `fs` (from node)... can you replace it and remove `fs-extra` ?
> 
> You did a great job so far! The community really appreciates it!

:) I saw it too yesterday, only basic `fs` methods except these two `ensureDir` as you say ; so I let it go.

But I went back to it today as you asked for: `fs` for everything + install and use of `mkdirp` for this precise need (a small and single-responsibility package ; thought about `make-dir` too but chose this one).

**Comment by rodrigograca31** [comment]
I will now tag it as 4.0.0 but cant push to npm yet....

**Comment by rodrigograca31** [comment]
`4.0.0` was published to npm! :partying_face: :tada: 

(will now add `"node": ">=10"`)

**Comment by pklaes** [comment]
Great! Thanks a lot, guys!

**Pull request merged**